### DISCUSSION TITLE
Lingering Issue: Could not fetch Health, TLS and Validations status

### DISCUSSION THREAD

**Discussion by kencieszykowski** [kiali/kiali#4461] [Q&A]
Hey everyone!

Made a post a few weeks back about some issues with Kiali, and are having some lingering issues around still, preventing some things from being displayed.  

Environment:
EKS // Kubernetes v1.18
Kiali // v1.37.0
Kiali Operator // v1.37.0
Istio // v1.10
Both Istio and Kiali are deployed using respective CRs and Operators

[More configuration can be found in this thread](https://github.com/kiali/kiali/discussions/4402)

The problem is that we're getting the following lines from Kiali's dashboard (this has always happened):
![capture 2021-10-28 at 4 46 23 PM](https://user-images.githubusercontent.com/13755642/139351625-dedbf0eb-d0e7-4b56-ab3b-1a8d49dfdcf7.png)

Originally, we didn't think very much of it; but now that we're pushing over to Prod the lack of visibility on these things is problematic.  In addition, I don't see the items in the screenshots in logs whatsoever (grep for TLS/Health/Validation).  There are these really strange logs, however:

```
2021-10-28T23:11:42Z ERR Failing starting the port forwarding
2021-10-28T23:11:42Z ERR Error fetching config_map: unable to listen on any of the requested ports: [{15000 15000}]
2021-10-28T23:11:42Z ERR unable to listen on any of the requested ports: [{15000 15000}]
Unable to listen on port 15000: Listeners failed to create with the following errors: [unable to create listener: Error listen tcp4 127.0.0.1:15000: bind: address already in use unable to create listener: Error listen tcp6 [::1]:15000: bind: cannot assign requested address]
2021-10-28T23:11:44Z ERR Failing starting the port forwarding
2021-10-28T23:11:44Z ERR Error fetching config_map: unable to listen on any of the requested ports: [{15000 15000}]
2021-10-28T23:11:44Z ERR unable to listen on any of the requested ports: [{15000 15000}]
```
These immediately follow the logs about being unable to find Grafana-- since we aren't using Grafana, those lines are to be expected, but the above lines are not.  Port 15000 is clear on our firewall between these instances and the control plane, and the only place I see 15000 configured anywhere is the Istio Admin Port.

In addition to the strange logs above, we're getting this for EVERY namespace:
```
2021-10-28T23:57:27Z ERR Workload istio-ns-injector-1635462000 is not found as Job
```

From what I've read; Jobs seem to be a default exclusion for the cache; yet here it is, trying to find them.  

Lastly: We send Envoy traces over to Datadog (via specifying the Tracing config in Istio to be the built-in Datadog configuration); there is no way to retrieve these traces/spans for usage in Kiali since it only supports Jaeger for displaying these, right?  

Any help would be appreciated; this seems like strange behavior.

**Comment by israel-hdez** [comment]
Usually, these error notifications have more details about. Strangely, in you screenshots I see there is nothing after the colons.

Perhaps, can you use the developer tools of the browser and find the failing requests? Maybe you can extract the response of any failing request and post them here.

**Comment by xeviknal** [comment]
Hi @kencieszykowski, 
I am late to the party, but let me add some ideas.

Istio has the following port requirements: https://istio.io/latest/docs/ops/deployment/requirements/#ports-used-by-istio
The port `15000` is one of the required ones used. It is opened (locally) by the sidecar (envoy) for administrative purposes.
Kiali opens a port in its pod with destination to one sidecar. In order to reach that port, kiali needs to do a port-forward.
It looks like kiali has already opened a that port before.

One question, after restarting the kiali pod, does this error appear anymore? Is it possible that the problem arise after some time using kiali (minutes, hours, days)? Or is it appearing with a fresh kiali pod?
